pkg/api/validation/validation.go

```diff
@@ -397,6 +401,20 @@ func validateISCSIVolumeSource(iscsi *api.ISCSIVolumeSource) errs.ValidationErro
 	return allErrs
 }
 
+func validateFCVolumeSource(fc *api.FCVolumeSource) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if len(fc.TargetWWNs) < 1 {
+		allErrs = append(allErrs, errs.NewFieldRequired("targetWWNs"))
+	}
+	if fc.FSType == "" {
+		allErrs = append(allErrs, errs.NewFieldRequired("fsType"))
+	}
+	if fc.Lun < 0 || fc.Lun > 255 {
```

Comment:
10 bucks says It fails ONLY because kubectl --validate is set by default.
If you unset that the apiserver will accept it (because it unmarshals
before validation and Go defaults to 0).

If this is a required field and default=0 is not correct, it needs to be a
pointer in the verisoned API (but not the internal)

On Wed, Sep 16, 2015 at 9:05 AM, Huamin Chen <notifications@github.com>
wrote:

> In pkg/api/validation/validation.go
> <https://github.com/kubernetes/kubernetes/pull/12599#discussion_r39649342>
> :
>
> > @@ -397,6 +401,20 @@ func validateISCSIVolumeSource(iscsi *api.ISCSIVolumeSource) errs.ValidationErro
> >  	return allErrs
> >  }
> >
> > +func validateFCVolumeSource(fc *api.FCVolumeSource) errs.ValidationErrorList {
> > +	allErrs := errs.ValidationErrorList{}
> > +	if len(fc.TargetWWNs) < 1 {
> > +		allErrs = append(allErrs, errs.NewFieldRequired("targetWWNs"))
> > +	}
> > +	if fc.FSType == "" {
> > +		allErrs = append(allErrs, errs.NewFieldRequired("fsType"))
> > +	}
> > +	if fc.Lun < 0 || fc.Lun > 255 {
>
> @thockin <https://github.com/thockin> it is required, so you have to
> specify it. But if you specify negative numbers or higher than 255, this
> line will reject you.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12599/files#r39649342>.
>